addons/account/models/account_move.py

```diff
@@ -2918,6 +2918,40 @@ def is_internal_partner(partner):
         move.message_subscribe(list(all_followers_ids))
         return move
 
+    def _post_check_invoice_non_negative(self):
+        """It checks that the total amount of a normal account cannot be less than zero.
+
+        In some accounting localizations this should be adjusted/disabled.
+        """
+        for move in self:
+            if (
+                move.is_invoice(include_receipts=True)
+                and float_compare(move.amount_total, 0.0, precision_rounding=move.currency_id.rounding) < 0
+            ):
+                raise UserError(_("You cannot validate an invoice with a negative total amount. "
+                                  "You should create a credit note instead. Use the action menu to transform it into "
+                                  "a credit note or refund."))
```

Comment:
I have no issue to implement this as a `api.constrains`, but I had adding seemingly useless hooks everywhere.

Reply:
In Hungarian invoicing, issuing a normal invoice with a negative sign is a mandatory feature that Odoo does not currently offer, because with the current _post() implementation i cannot do.

So i really don't care how this will be implemented, what i really want is for the Hungarian invoicing i must disable this check.

What you suggest, should i recode this as a `api.constraint`, or leave it as a method what any l10n module can owerwrite?

Reply:
yes